### ISSUE TITLE
node-webcrypto-ossl node-gyp rebuild problem 

### ISSUE THREAD

**Issue by goriunov** [metaplex-foundation/metaplex#1672]
**Describe the bug**

@yusefnapora  

After the upgrade of this pr https://github.com/metaplex-foundation/metaplex/commit/04d3eb9883272f92fde2bc894e585e417f880384 on master branch, which have pretty much updated lots of deps on yarn lock, users need to have additional utilities to be able to perform yarn install.

Here is example output on windows when running yarn install if there is no additional bindings building tools like python etc... specifically module node-webcrypto-ossl is casing this issue, which is interesting as that module is depricated already and not sure why it is present in the repo.

I am working through the issue and tracking the deps as of right now. As i believe there should not be any install errors like that for end user.

```
error C:\Users\goriunov\Desktop\github\metaplex-latest\js\node_modules\node-webcrypto-ossl: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments: 
Directory: C:\Users\goriunov\Desktop\github\metaplex-latest\js\node_modules\node-webcrypto-ossl
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@5.1.1
gyp info using node@16.13.1 | win32 | x64
gyp info find Python using Python version 3.7.9 found at "C:\Python\Python3.7\python.exe"
gyp ERR! find VS 
gyp ERR! find VS msvs_version not set from command line or npm config
gyp ERR! find VS VCINSTALLDIR not set, not running in VS Command Prompt
gyp ERR! find VS could not use PowerShell to find Visual Studio 2017 or newer
gyp ERR! find VS looking for Visual Studio 2015
gyp ERR! find VS - not found
gyp ERR! find VS not looking for VS2013 as it is only supported up to Node.js 8
gyp ERR! find VS
gyp ERR! find VS **************************************************************
gyp ERR! find VS You need to install the latest version of Visual Studio
gyp ERR! find VS including the "Desktop development with C++" workload.
gyp ERR! find VS For more information consult the documentation at:
gyp ERR! find VS https://github.com/nodejs/node-gyp#on-windows
gyp ERR! find VS **************************************************************
gyp ERR! find VS
gyp ERR! configure error
gyp ERR! stack Error: Could not find any Visual Studio installation to use
gyp ERR! stack     at VisualStudioFinder.fail (C:\Users\goriunov\Desktop\github\metaplex-latest\js\node_modules\node-gyp\lib\find-visualstudio.js:121:47)   
gyp ERR! stack     at C:\Users\goriunov\Desktop\github\metaplex-latest\js\node_modules\node-gyp\lib\find-visualstudio.js:74:16
gyp ERR! stack     at VisualStudioFinder.findVisualStudio2013 (C:\Users\goriunov\Desktop\github\metaplex-latest\js\node_modules\node-gyp\lib\find-visualstudio.js:351:14)
gyp ERR! stack     at C:\Users\goriunov\Desktop\github\metaplex-latest\js\node_modules\node-gyp\lib\find-visualstudio.js:70:14
gyp ERR! stack     at C:\Users\goriunov\Desktop\github\metaplex-latest\js\node_modules\node-gyp\lib\find-visualstudio.js:372:16
gyp ERR! stack     at C:\Users\goriunov\Desktop\github\metaplex-latest\js\node_modules\node-gyp\lib\util.js:54:7
gyp ERR! stack     at C:\Users\goriunov\Desktop\github\metaplex-latest\js\node_modules\node-gyp\lib\util.js:33:16
gyp ERR! stack     at ChildProcess.exithandler (node:child_process:404:5)
gyp ERR! stack     at ChildProcess.emit (node:events:390:28)
gyp ERR! stack     at maybeClose (node:internal/child_process:1064:16)
gyp ERR! System Windows_NT 10.0.19043
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\goriunov\\Desktop\\github\\metaplex-latest\\js\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\goriunov\Desktop\github\metaplex-latest\js\node_modules\node-webcrypto-ossl
```

**Comment by goriunov** [comment]
![image](https://user-images.githubusercontent.com/18750503/151657847-046d3cbf-b3c9-44b3-a800-6044ab5fd385.png)

**Comment by goriunov** [comment]
After checking out the deps three the reason why things are failing is due to the minor version upgrade on redstone-node dep, looks like they have added streamr-client dep which is bringing in node-webcrypto-ossl the dep which requires compilation tools:


https://github.com/redstone-finance/redstone-node/commit/8dd07c3ffaa73d09fdaf94c777ad98544e533a64#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519

**Comment by Chrissyticky** [comment]
I'm having this exact same issue.

**Comment by TheMarketfoxx** [comment]
download the 2017 version of vs then install the desktop c++ files once completed it should work. https://code.visualstudio.com/updates/v1_17.

**Comment by goriunov** [comment]
@TheMarketfoxx the dep requiring the build is not really needed for the CM itself.. Also that dep is deprecated and the underline dep is using it for some reason... the pr above reverts that dependency which got upgraded by mistake. I mean installing visual studio etc will help, it is just unnecessary headache :)

**Comment by TheMarketfoxx** [comment]
> 

understood

**Comment by Deliceli58** [comment]
I'm having this exact issue. But I don't understand what should I do @goriunov can you please clarify Im newbie in coding
